src/pages/sighting/identification/CustomMatchingSetForm.jsx

```diff
@@ -31,7 +31,13 @@ export default function CustomMatchingSetForm({
       algorithms,
       matching_set: buildMatchingSetQuery(regionSchema, region),
     });
-  }, [algorithms?.length, region]);
+  }, [
+    setIdConfig,
```

Comment:
this is the one. it's because `setIdConfig` is a setstate passed in from the parent function, so the linter's not aware of it

Reply:
Is there harm to including it?

Reply:
That way, we could avoid the cognitive labor of repeatedly having to figure out that we should ignore this particular warning.

Reply:
just checked, there doesn't seem to be any harm in including it